### ISSUE TITLE
Re-project embeddings with same size?

### ISSUE THREAD

**Issue by helpmefindaname** [flairNLP/flair#1830]
Hello, based on the sequencetagger [here](https://github.com/flairNLP/flair/blob/master/flair/models/sequence_tagger_model.py#L172) and [here](https://github.com/flairNLP/flair/blob/master/flair/models/sequence_tagger_model.py#L652) is a retransforming of the embeddings, by applying a linear layer without activation function. 

As far as I understand, this layer can be merged with the following layer:
e.g. for input X, the reprojection T with parameters U and b will be followed by a lstm/gru/linear layer to produce the next output O. (I use the linear layer for simplification with parameters V and c)
T = U * x + b
O = V * T + c

This can be reduced to O = W * x + b' where W = U * V and b' = c + V * b and essentially doesn't change anything except for having less parameters to train.

For LSTM/GRU we basically have the same linear transformation, just that also another one for the hidden state too.

Hence my question is, what is the point of the retransforming as default parameter?
I get that choosing a smaller size reduces the rank of W and therefore enfoces different result, but by default it's the same size as the embedding. 

I'd be interested to hear any clever tricks that sound unintuitive to me

**Comment by alanakbik** [comment]
Hello @helpmefindaname we originally added this layer since in earlier Flair versions most embeddings were not fine-tuneable (see #690 for a more detailed discussion). Our classic word embeddings are still only implemented as a simple lookup table and so cannot be directly fine-tuned. The linear layer on top was intended to enable a transformation of the entire embedding space to alleviate this. It is probably no longer necessary if only embeddings are used that are directly fine-tuneable. 

I haven't run extensive experiments, but in our default setup (Flair embeddings without fine-tuning + word embeddings) it seems that it works better if this layer is left in. I'm not entirely sure why this is, perhaps there is some regularization effect or smoothing of the gradients.
